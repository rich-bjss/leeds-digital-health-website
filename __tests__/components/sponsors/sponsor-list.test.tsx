import { render, screen } from "@testing-library/react"

import SponsorList from "@/components/sponsors/sponsor-list"

import Sponsor from "@/lib/model/sponsor"
import { getResolvedAsyncComponent } from "@/__tests__/test-helper"

const TEST_SPONSOR_DATA: Sponsor[] = []

for (let i = 0; i < 8; i++) {
  TEST_SPONSOR_DATA.push({
    name: `sponsor-${i}`,
    logo: {
      url: `/logo-url-${i}`,
      description: `logo-description-${i}`
    },
    sys: {
      id: `sponsor-id-${i}`
    }
  })
}

jest.mock("../../../lib/api/sponsors", () => ({
  getAllSponsors: () => TEST_SPONSOR_DATA
}))

describe("SponsorList component", () => {
  it("includes the correct title", async()=>{
    const ResolvedSponsorList = await getResolvedAsyncComponent(SponsorList)
    const { container } = render(<ResolvedSponsorList />)

    const heading = screen.getByRole("heading", { level: 1 })

    expect(heading).toMatchSnapshot("The Leeds Digital Health Committee")
  })

  it("correctly renders a list of sponsor images from test data", async () => {
    const ResolvedSponsorList = await getResolvedAsyncComponent(SponsorList)
    render(<ResolvedSponsorList />)

    TEST_SPONSOR_DATA.forEach((sponsor) => {
      const image = screen.getByAltText(sponsor.name)
      expect(image).toBeInTheDocument()
    })
  })
})
