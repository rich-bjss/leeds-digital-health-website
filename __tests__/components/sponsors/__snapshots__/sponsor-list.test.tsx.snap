// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SponsorList component includes the correct title: The Leeds Digital Health Committee 1`] = `
<h1
  class="w-full text-center text-2xl sm:text-4xl font-bold text-gray-500"
>
  The Leeds Digital Health Committee
</h1>
`;
